### PR TITLE
add support for docker-based mongo

### PR DESCRIPTION
**Pull request by cmentzer**
To test: 

- pip install scooze
- install docker desktop and make sure it's running
- run `scooze setup docker` to pull and initailize a docker-based mongo instance
- populate that DB as normal with `scooze load-cards oracle`

Limitations: 
- Needs docker installed and running
- Needs port 27017 to be free on the host machine

### CODE REVIEW

File: src/scooze/cli.py [hunk 1]
```diff
@@ -26,6 +29,12 @@ def parse_args():
         f"""deck data to test with, or to run the scooze Swagger UI/ReDocs.\n"""
```
Comment (iambroadband) [inline src/scooze/cli.py:29]: Should we sort these alphabetically or something? Do we even care since #170 will take care of all of this anyway?

**Review by iambroadband — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -31,19 +31,31 @@ A flexible data layer for applications working with Magic: the Gathering cards,
...
+    - Using Docker:
 
-    Scooze depends on MongoDB to run your local database.
+        *Make sure [Docker Desktop](https://www.docker.com/products/docker-desktop/) is running.*
 
-    *You can use scooze without MongoDB if you don't intend to use any of its database-related features.*
+         ```
+         scooze setup docker
+         ```
+         * This will pull the latest MongoDB issued image for your machine's distribution, and spin up the container with port 27017 bound to the host's port 27017.
```
Comment (fjork3) [inline README.md:42]: You'll still need to load cards (on initial setup), so this should be going to (3) and not (4).

File: src/scooze/api/bulkdata.py [hunk 3]
```diff
@@ -20,6 +21,8 @@ async def load_card_file(file_type: ScryfallBulkFile, bulk_file_dir: str) -> Non
     """
 
     file_path = f"{bulk_file_dir}/{file_type}.json"
+    if not os.path.exists(bulk_file_dir):
```
Comment (fjork3) [inline src/scooze/api/bulkdata.py:24]: Do we need this here? `download_bulk_data_file` should already do this under the hood, and I don't think we need to make dirs unless we're actually loading the file.

File: src/scooze/cli.py (hunk 1 above)
Comment (fjork3) [reply src/scooze/cli.py:29]: The possible command line args? I think they're logically sorted as-is, happy enough letting it be until we start the overhaul.

**Comment by fjork3** [comment]
Base branch for this PR should be `dev` rather than `main`

File: src/scooze/api/bulkdata.py (hunk 3 above)
Comment (iambroadband) [reply src/scooze/api/bulkdata.py:21]: it does not do it under the hood

Comment (iambroadband) [reply src/scooze/api/bulkdata.py:21]: If the directory doesn't exist, it raises an Error here. How should we handle that before download_bulk_data_file runs?

Comment (fjork3) [reply src/scooze/api/bulkdata.py:21]: If it's raising an error that's not `FileNotFoundError`, we could catch it in the same except block underneath; https://github.com/arcavios/scooze/blob/5f0838d9bac8c9f37e270d78558a69de2af894f5/src/scooze/bulkdata.py#L29 should already handle making the appropriate dir.

**Pull request merged**